layers/meta-balena-isg-503/conf/samples/local.conf.sample

```diff
@@ -1,6 +1,8 @@
 # Supported machines
 #MACHINE ?= "isg-503"
 BALENA_STORAGE = "overlay2"
+# temporarily disable kernel headers in order to circumvent a compile issue of kernel-headers-test
+BALENA_DISABLE_KERNEL_HEADERS = "1"
```

Comment:
It take a lot of time to build.

Reply:
if I am not enable BALENA_DISABLE_KERNEL_HEADERS, it give a error,do you have idea for this?

_```
ERROR: kernel-headers-test-1.0-r0 do_compile: ExecutionError('/home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/temp/run.do_compile.3025566', 2, None, None)
ERROR: Logfile of failure stored in: /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/temp/log.do_compile.3025566
Log data follows:
| DEBUG: Executing shell function do_compile
| The command '/bin/sh -c ARCH=${kernel_arch} CROSS_COMPILE=${cross_compile_prefix} make -C /usr/src/app/kernel_source/*/build modules_prepare' returned a non-zero code: 2
| WARNING: exit code 2 from a shell command.
ERROR: Task (/home/rockemb/Balena/enpal/github/build/../layers/meta-balena/meta-balena-common/recipes-kernel/linux/kernel-headers-test.bb:do_compile) failed with exit code '1'
```_

Reply:
You can see more details of the error in the run.do_compile log for this package. Check out  /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/temp/log.do_compile

Reply:
DEBUG: Executing shell function do_compile
The command '/bin/sh -c ARCH=${kernel_arch} CROSS_COMPILE=${cross_compile_prefix} make -C /usr/src/app/kernel_source/*/build modules_prepare' returned a non-zero code: 2
WARNING: exit code 2 from a shell command.

Reply:
this is all in log.do_compile.3025566

Reply:
Sorry, I said run.do_compile but I pasted the wrong path
Try /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/temp/run.do_compile

Reply:
Yes , I will try this command  and add more logger in Dockerfile

# line: 32, file: /home/rockemb/Balena/enpal/github/build/../layers/meta-balena/meta-balena-common/recipes-kernel/linux/kernel-headers-test.bb
do_compile() {
    rm -rf /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work
    mkdir -p /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work
    cp /home/rockemb/Balena/enpal/github/build/tmp/deploy/images/isg-503/kernel_source.tar.gz /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work
    cp /home/rockemb/Balena/enpal/github/build/tmp/deploy/images/isg-503/kernel_modules_headers.tar.gz /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work
    cp "/home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0"/Dockerfile /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work/
    cp -r "/home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0"/example_module /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work/

    IMAGE_ID=$(DOCKER_API_VERSION=1.22 docker build --build-arg kernel_arch=arm64 --build-arg cross_compile_prefix=aarch64-linux-gnu- /home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0/work)
    # We don't pipe in previous line so that we can catch errors.
    IMAGE_ID=$(echo "$IMAGE_ID" | grep -o -E '[a-z0-9]{12}' | tail -n1)
    DOCKER_API_VERSION=1.22 docker rmi "$IMAGE_ID"
}

cd '/home/rockemb/Balena/enpal/github/build/tmp/work/armv8a-poky-linux/kernel-headers-test/1.0-r0/kernel-headers-test-1.0'
do_compile

# cleanup
ret=$?
trap '' 0
exit $ret

Reply:
need python3
/usr/bin/env: 'python3': No such file or directory
add "apt install python3" in the dockfile will be fine
If I change the source code (kernel/build) use python instead of python3. some OS (for example ubuntu 20.04 have not python )will fail

Reply:
> need python3 /usr/bin/env: 'python3': No such file or directory add "apt install python3" in the dockfile will be fine If I change the source code (kernel/build) use python instead of python3. some OS (for example ubuntu 20.04 have not python )will fail

@RockSupport the commit that introduces this problem is https://github.com/JeffyCN/meta-rockchip/commit/1258d9241edf512b0d10076896fd5ce36ba58cb4.

Any idea why the linux recipe modifies the kernel scripts in that way?